examples/schema_template_contract/main.tf

```diff
@@ -55,52 +50,48 @@ resource "mso_schema_template_filter_entry" "filter_entry_2" {
   display_name         = "Filter2"
   entry_name           = "entry2"
   entry_display_name   = "entry2"
-  entry_description    = "DemoEntry"
-  ether_type           = "arp"
-  ip_protocol          = "eigrp"
-  tcp_session_rules    = ["acknowledgement"]
-  destination_from     = "unspecified"
-  destination_to       = "unspecified"
-  source_from          = "unspecified"
-  source_to            = "unspecified"
-  arp_flag             = "unspecified"
-  stateful             = false
-  match_only_fragments = false
 }
 
 resource "mso_schema_template_contract" "template_contract" {
-  schema_id     = mso_schema_template_filter_entry.filter_entry_2.schema_id
-  template_name = mso_schema_template_filter_entry.filter_entry_2.template_name
+  schema_id     = mso_schema_template_filter_entry.filter_entry.schema_id
+  template_name = mso_schema_template_filter_entry.filter_entry.template_name
   contract_name = "Contract1"
   display_name  = "Contract1"
   filter_type   = "bothWay"
+  target_dscp   = "af11"
+  priority      = "level1"
   scope         = "context"
   filter_relationship {
-    filter_schema_id     = mso_schema_template_filter_entry.filter_entry.schema_id
-    filter_template_name = mso_schema_template_filter_entry.filter_entry.template_name
     filter_name          = mso_schema_template_filter_entry.filter_entry.name
+    filter_type          = "bothWay"
+    action               = "permit"
+    priority             = "default"
+    directives           = ["log", "no_stats"]
   }
   filter_relationship {
     filter_schema_id     = mso_schema_template_filter_entry.filter_entry_2.schema_id
     filter_template_name = mso_schema_template_filter_entry.filter_entry_2.template_name
     filter_name          = mso_schema_template_filter_entry.filter_entry_2.name
+    filter_type          = "bothWay"
+    action               = "deny"
+    priority             = "level2"
+    directives           = ["log"]
```

Comment:
can we have one filter_relationship with just name and type(required params) and and another filter_relationship{} with all the suboptions it has?

Reply:
Yes we should be able to but noticed a bug in the logic. Will adjusted code to set correctly.